activerecord/lib/active_record/transactions.rb

```diff
@@ -479,7 +479,7 @@ def has_transactional_callbacks?
       # the TransactionState, and rolls back or commits the Active Record object
       # as appropriate.
       def sync_with_transaction_state
-        if (transaction_state = @transaction_state)&.finalized?
```

Comment:
Why not write 
```suggestion
return unless @transaction_state&.finalized?
if transaction_state = @transaction_state
```
 If we use this no need to touch the remaining 13 places right?

Reply:
Because that’s exactly the same as the previous code and this wouldn’t give any performance boost. The boost comes from the transaction state being false for the majority of calls.